### PR TITLE
Feature/302 clickable tooltips

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [EQ-302](https://jira.epa.gov/browse/EQ-302)

## Main Changes:
* Dropped the `@reach/tooltip` dependency, and switched to a version of the USWDS tooltip adapted for React.
  * This version draws from the USWDS source code as well as another adaptation from [uswds-react](https://github.com/trussworks/react-uswds/tree/main/src/components/Tooltip).
* Fixed a field label accessibility issue, where the name presented to a screen reader included tooltip and input content.

## Steps To Test:
1. Test the tooltips on a variety of screen sizes. Verify that touching the tooltip trigger on a touchscreen device or emulator opens the tooltip, and that clicking the trigger on desktop does not close the tooltip. Additionally, focusing the trigger with the keyboard should open the tooltip.

## N.B.
When testing the tooltips in the browser's mobile emulator at <= 360px, I found that some long tooltips (e.g. _Associated Action ID (Name))_ could not be positioned properly, and a warning is printed to the console in this case. However, the same tooltip displays fine on my iPhone and on a very narrow desktop window, so I believe the issue is related to the calculation of `win.innerHeight` or `document.documentElement.clientHeight` when using the emulator, and can be disregarded.

### CODE REVIEW

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Works well! 

I noticed we were missing types for the `classnames` package, so I went ahead and added them in. 

I did notice the same Chrome emulator issues on my Android phone. I included a possible solution to that issue below. Can you try it out on your iPhone?

File: app/client/src/components/tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+  const rect = el.getBoundingClientRect();
+  const win = window;
+  const docEl = document.documentElement;
+
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (win.innerHeight || docEl.clientHeight) &&
+    rect.right <= (win.innerWidth || docEl.clientWidth)
+  );
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/tooltip.tsx:261]: Thanks for checking it on an Android phone. I tested the fix on an iPhone and it works well.

File: app/client/src/components/tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+## Types
+*/
+
+type InfoTooltipProps = Omit<TooltipProps, 'children'> & {
+  description?: string;
+};
+
+type TooltipProps = {
+  text: string;
+  position?: 'top' | 'bottom' | 'left' | 'right';
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/tooltip.tsx]: Sounds good. I suppose it allows for a preference, but we always use the default 'top' value.

File: app/client/src/components/tooltip.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/tooltip.tsx]: I don't think it fixes the opposite problem, though, when at the very top of the screen. I'll try to adjust for that case.

**Review by cschwinderg — APPROVED** [review]
The fix works well! Thank you!

**Pull request merged**